src/videojs-hls.js

```diff
@@ -286,7 +286,18 @@ videojs.Hls.prototype.handleSourceOpen = function() {
  */
 videojs.Hls.prototype.play = function() {
   if (this.ended()) {
-    this.mediaIndex = 0;
+    // When the video ends and the 'ended' event fires, current time is set to 0. If the video is in this state, or the
+    // current time is equal to the duration (ie, play head is at the end position), reset the video to the beginning on
+    // a play event.
+    if (this.currentTime() === 0 || this.currentTime() === this.duration()) {
```

Comment:
My reading of [the ended property](https://html.spec.whatwg.org/multipage/embedded-content.html#dom-media-ended) seems to imply that `currentTime` should never be zero at this point. I think we should fix that rather than have all of these cases to handle here.